learn/using-the-openapi-tools.md

```diff
@@ -28,91 +28,104 @@ Code generation from OpenAPI to Ballerina can produce `ballerina mock services`
 
 For build time client stub generation, annotation support is provided.
 
-- [Mock Service from OpenAPI](#mock-service-from-openapi)
-- [Client Stub from OpenAPI](#client-stub-from-openapi)
-- [Service to OpenAPI Export](#service-to-openapi-export)
-- [Client Stub for Service](#client-stub-for-service)
+- [OpenAPI to Ballerina](#openAPI-to-ballerina)
+    - [Mock Service from OpenAPI](#mock-service-from-openapi)
+    - [Client Stub from OpenAPI](#client-stub-from-openapi)
+- [Ballerina to OpenAPI](#ballerina-to-openAPI)    
+    - [Service to OpenAPI Export](#service-to-openapi-export)
+- [Client Stub for Service](#client-stub-for-service)    
 - [Samples](#samples)
     - [Mock Service from OpenAPI Sample](#mock-service-from-openapi-sample)
     - [Client Stub from OpenAPI Sample](#client-stub-from-openapi-sample)
     - [OpenAPI from Service Sample](#openapi-from-service-sample)
     - [Client Stub from Service Sample](#client-stub-from-service-sample)
+- [OpenAPI Validator Compiler Plugin](#openAPI-validator-compiler-plugin)
+     
 
-## Mock Service from OpenAPI
-
+###OpenAPI to Ballerina
+####Mock service and Client stub from OpenAPI
 ```
-ballerina openapi gen-service <moduleName>:<serviceName> 
-                               <openapi_contract>
-                               [-c: copy-contract] 
-                               [-o: outputFile]
+ballerina openapi   -i <openapi-contract> 
+               [--service-name: generated files name]
+               [--tags: tags list]
+               [--operations: operationsID list]
+               [(-o|--output): outputFile]
 ```
+Generates both Ballerina service and Ballerina client stub for a given OpenAPI file. 
+
+This parameter `-i <openapi-contract>` of the command is mandatory. As an input it will take the path to the resource file. 
 
-Generates a Ballerina service for the OpenAPI file.
+`--service-name`  This is an optional parameter. If you are interested in naming generated files by given name, 
+you can use this parameter with command. 
 
-This generated service is a mock version of the actual Ballerina service. Generated sources contain the service definition in `src/<module-name>/` and the OpenAPI contract that used to generate will be copied to `src/<module-name>/resources`. 
+You can give the specific tags and operations that you need to document as services without documenting the 
+whole operation using these optional commands `--tags` and `--operations`.
 
-## Client Stub from OpenAPI
+`(-o|--output)` is the optional parameter. You can give the specific direction path to generate the files. 
+If not it will take the execution path as the target path.
```

Comment:
```suggestion
`(-o|--output)` is an optional parameter. You can give the output path for the generated files. 
If not it will take the execution path as the output path.
```